x/merkledb/proof.go

```diff
@@ -45,6 +53,68 @@ type ProofNode struct {
 	Children    map[byte]ids.ID
 }
 
+// Assumes [node.Key.KeyPath.NibbleLength] <= math.MaxUint32.
```

Comment:
Changed this to `uint64` (proto uses varints so this doesn't actually make the message larger on the wire.) 

This actually isn't _really_ enforced. Like, our assumption here is that there is never a key whose length is more than half of `math.MaxUint64`. We could add checks in the code to ensure this is upheld but I talked with @StephenButtolph and he (and I) are OK with omitting those since, well, that's a really big number.